src/Event/Http/HttpRequestEvent.php

```diff
@@ -346,4 +342,17 @@ public function isFormatV2(): bool
     {
         return $this->payloadVersion === 2.0;
     }
+
+    /**
+     * When keys within a URL query string contain dots, PHP's parse_str() method
+     * converts them to underscores. This method works around this issue so the
+     * requested query array returns the proper keys with dots.
+     *
+     * @return array<string, string>
+     */
+    private function queryStringToArray(string $query): array
+    {
+        $queryString = Query::fromString($query);
+        return $queryString->toArray();
```

Comment:
We could maybe make this easier
```suggestion
        return Query::fromString($query)->toArray();
```